### ISSUE TITLE
Containerd: Leverage Privileged Containers Enhancements

### ISSUE THREAD

**Issue by matthewpereira** [concourse/concourse#5713]
### TLDR

- The `containerd` runtime should make image building faster via building images without `privileged: true` 

### Problem

- When utilizing the current `garden` runtime, there is a lot of performance overhead when using `privileged: true`  for building images which can result in slow initialization times and overall poor performance. 
- There wasn't a great solution to this problem when leveraging just `garden` as the container runtime.  

### Proposed Solution

- Add `containerd` as a secondary runtime solution along with `garden` ✅
- Utilize `containerd` to make image building faster via one of the following options:
    * Making it so Concourse can build images without `privileged: true` 
    * Eliminating the performance overhead of using `privileged: true` 
- Ensure users can still run containers-in-containers while taking this approach

### Outcomes

- Faster initialization times for workloads involving image building

### Current Status

- [`containerd` GA'd as of v7.0.0](https://github.com/concourse/concourse/releases#containerd) 🎉
- In Progress

### References
#5679 - Transition to `containerd`
#6475 - Define what privileged means in context to `containerd`

**Comment by taylorsilva** [comment]
Closing because containerd is about to become the default runtime #7106. Privileged container were sped up by changes to baggageclaim when using overlay #6620 which is the recommended driver linux workers should use.
